src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -2,45 +2,551 @@
+            return result;
+        }
+
+        public static unsafe Vector256<byte> CreateScalar(byte value)
+        {
+            var result = Vector256<byte>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<byte>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<double> CreateScalar(double value)
+        {
+            var result = Vector256<double>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<double>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<short> CreateScalar(short value)
+        {
+            var result = Vector256<short>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<short>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<int> CreateScalar(int value)
+        {
+            var result = Vector256<int>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<int>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<long> CreateScalar(long value)
+        {
+            var result = Vector256<long>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<long>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector256<sbyte> CreateScalar(sbyte value)
+        {
+            var result = Vector256<sbyte>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<sbyte>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<float> CreateScalar(float value)
+        {
+            var result = Vector256<float>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<float>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector256<ushort> CreateScalar(ushort value)
+        {
+            var result = Vector256<ushort>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<ushort>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector256<uint> CreateScalar(uint value)
+        {
+            var result = Vector256<uint>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<uint>, byte>(ref result), value);
+            return result;
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector256<ulong> CreateScalar(ulong value)
+        {
+            var result = Vector256<ulong>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector256<ulong>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector256<T> CreateScalar<T>(Vector128<T> value) where T : struct
```

Comment:
What is the purpose of this API?

Reply:
This will be explicitly brought up at the API review (still waiting for confirmation on the date from @terrajobst) and we can decide if `ToVector256` or `AsVector256` (and the `Unsafe` counterpart) makes more sense. If we decide to expose it as an instance method, we will also need to expect that users would know to look for them there, when the `Vector256<T> Create<T>(Vector128<T> lower, Vector128<T> upper)` method exists elsewhere.

Reply:
I do not think `CreateScalar` expressing the semantic. Perhaps, we can name them to `ZeroExtendToVector256/128` for "safe" versions and other names (i.e., `UnsafeExtendToVector256/128`) for the "unsafe" version.
BTW, I am not sure if it worthwhile to expose an `UnsafeExtendToVector256`. On x86, `ZeroExtendToVector256` is just a zero-latency instruction `vmovaps xmm, xmm`.